src/main/scala/edu/twitter/model/GradientBoostingModel.scala

```diff
@@ -0,0 +1,100 @@
+package edu.twitter.model
+
+import org.apache.spark.mllib.regression.LabeledPoint
+import org.apache.spark.mllib.tree.GradientBoostedTrees
+import org.apache.spark.mllib.tree.configuration.BoostingStrategy
+import org.apache.spark.mllib.tree.model.GradientBoostedTreesModel
+import org.apache.spark.rdd.RDD
+
+/**
+  * Build and evaluate a gradient boosting model from training and testing data set.
+  * @param trainingData
+  * @param validationData
+  */
+class GradientBoostingModel(trainingData: RDD[LabeledPoint], validationData: RDD[LabeledPoint]) {
+
+  /**
+    * Create a GradientBoosting Classification model using a Gradient Boosting model.
+    * The reason we chose Gradient Boosting for classification over some other model
+    * is because it’s easy to use (doesn’t require tons of parameter tuning), and it
+    * tends to have a high classification accuracy. For this reason it is frequently
+    * used in machine learning competitions.
+    * The tuning parameters we’re using here are:
+        -number of iterations (passes over the data)
+        -Max Depth of each decision tree
+    * @return sentiment classification model
+    */
+  def getModel(): GradientBoostedTreesModel = {
+    val boostingStrategy = BoostingStrategy.defaultParams("Classification")
+    boostingStrategy.setNumIterations(20) //number of passes over our training data
+    boostingStrategy.treeStrategy.setNumClasses(2) //We have two output classes: happy and sad
+    boostingStrategy.treeStrategy.setMaxDepth(5)
+    //Depth of each tree. Higher numbers mean more parameters, which can cause overfitting.
+    //Lower numbers create a simpler model, which can be more accurate.
+    //In practice you have to tweak this number to find the best value.
+
+    var model = GradientBoostedTrees.train(trainingData, boostingStrategy)
+    evaluate(model)
```

Comment:
Small note here, can we separate the evaluation from the building. meaning we don't have to call this method here in the construction of the model.

Reply:
What I'm thinking about is to make this method return pair model and evaluation but I found our evaluation is only a console print, It doesn't make sense for me to make a separate class for evaluation because GradientBoostingModel doesn't extend anything like Model class or something, can we return model and evaluation?